projects/packages/jetpack-mu-wpcom/src/features/custom-css/custom-css.php

```diff
@@ -0,0 +1,1249 @@
+					),
+				)
+			);
+		}
+
+		/**
+		 * Check whether there are CSS Revisions for a given theme.
+		 *
+		 * Going forward, there should always be, but this was necessitated
+		 * early on by https://core.trac.wordpress.org/ticket/30854
+		 *
+		 * @param string $stylesheet Stylesheet name.
+		 *
+		 * @return bool|null|WP_Post
+		 */
+		public static function are_there_css_revisions( $stylesheet = '' ) {
+			$post = wp_get_custom_css_post( $stylesheet );
+			if ( empty( $post ) ) {
+				return $post;
+			}
+			return (bool) wp_get_post_revisions( $post );
+		}
+
+		/**
+		 * Core doesn't have a function to get the revisions url for a given post ID.
+		 *
+		 * @param string $stylesheet Stylesheet name.
+		 *
+		 * @return null|string|void
+		 */
+		public static function get_revisions_url( $stylesheet = '' ) {
+			$post = wp_get_custom_css_post( $stylesheet );
+
+			// If we have any currently saved customizations...
+			if ( $post instanceof WP_Post ) {
+				$revisions = wp_get_post_revisions( $post->ID, array( 'posts_per_page' => 1 ) );
+				if ( empty( $revisions ) || is_wp_error( $revisions ) ) {
+					return admin_url( 'themes.php?page=editcss' );
+				}
+				$revision = reset( $revisions );
+				return get_edit_post_link( $revision->ID );
+			}
+
+			return admin_url( 'themes.php?page=editcss' );
+		}
+
+		/**
+		 * Get a map of all theme names and theme stylesheets for mapping stuff.
+		 *
+		 * @return array
+		 */
+		public static function get_themes() {
+			$themes = wp_get_themes( array( 'errors' => null ) );
+			$all    = array();
+			foreach ( $themes as $theme ) {
+				$all[ $theme->name ] = $theme->stylesheet;
+			}
+			return $all;
+		}
+
+		/**
+		 * When we need to get all themes that have Custom CSS saved.
+		 *
+		 * @return array
+		 */
+		public static function get_all_themes_with_custom_css() {
+			$themes     = self::get_themes();
+			$custom_css = get_posts(
+				array(
+					'post_type'   => 'custom_css',
+					'post_status' => get_post_stati(),
+					'number'      => -1,
+					'order'       => 'DESC',
+					'orderby'     => 'modified',
+				)
+			);
+			$return     = array();
+
+			foreach ( $custom_css as $post ) {
+				$stylesheet = $post->post_title;
+				$label      = array_search( $stylesheet, $themes, true );
+
+				if ( ! $label ) {
+					continue;
+				}
+
+				$return[ $stylesheet ] = array(
+					'label' => $label,
+					'post'  => $post,
+				);
+			}
+
+			return $return;
+		}
+
+		/**
+		 * Handle the registering of admin scripts
+		 */
+		public static function wp_admin_enqueue_scripts() {
+			Assets::register_script(
+				'jetpack-customizer-css',
+				'../../build/core-customizer-css/core-customizer-css.js',
+				__FILE__,
+				array(
+					'dependencies' => array(
+						'jquery',
+						'customize-controls',
+						'underscore',
+					),
+					'in-footer'    => true,
+					'css_path'     => '../../build/customizer-control/customizer-control.css',
+				)
+			);
+
+			Assets::register_script(
```

Comment:
I'm not quite sure under which condition this script is loaded. I haven't been able to see it in the network panel during my testing.

Reply:
I'm also not sure, I've not been able to work out when it should be applied (given that it has `customize-selective-refresh` as a dependency), so would err on the side of making sure we leave it. Perhaps someone who has worked on Custom CSS historically may have some ideas on it though.